pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -914,6 +925,34 @@ type PrometheusSpec struct {
 	TSDB TSDBSpec `json:"tsdb,omitempty"`
 }
 
+type WhenScaledRetentionType string
+
+var (
+	WhenScaledRetentionTypeRetain WhenScaledRetentionType = "retain"
+	WhenScaledRetentionTypeDelete WhenScaledRetentionType = "delete"
+)
+
+type ShardRetentionPolicy struct {
+	// WhenScaled defines the retention policy when the Prometheus Server is scaled-down.
```

Comment:
Should it be named `WhenScaleDown`  if it only applies to scale down events?

Reply:
good point!